include/armadillo_bits/arma_forward.hpp

```diff
@@ -291,6 +292,12 @@ struct hdf5_name
     : filename(in_filename)
     , dsname  (in_dsname  )
     {}
+  inline
+  hdf5_name(const std::string& in_filename, const std::string& in_dsname, const bool& in_append)
+    : filename(in_filename)
+    , dsname  (in_dsname  )
+    , append  (in_append  )
```

Comment:
I wonder, could you just use a default argument of `false` for `in_append` here and reduce the number of constructors?  I think even the `in_dsname` argument could have a default too.